spec/project_spec.rb

```diff
@@ -23,6 +23,21 @@
     end
   end
 
+  describe 'cop message' do
+    let(:cops) { RuboCop::Cop::Cop.all }
+
+    it 'end with a period or a question mark' do
+      cops.each do |cop|
+        begin
+          msg = cop.const_get(:MSG)
+        rescue NameError
+          next
+        end
+        expect(msg).to match(/\.|\?|(?:\[.+\])|%s$/)
```

Comment:
Thanks your suggestion!

> This regular expression doesn't test that the message actually ends with the wanted punctuation, only that it contains it.

Oh, it hasn't worked. I'll fix it :+1: 

> And the ?: operator seems to be used incorrectly. 

Hmm, I don't think so. the `(?:...)` operator is non-captured group. https://ruby-doc.org/core-2.3.1/Regexp.html#class-Regexp-label-Grouping
It is used correctly.


I'll create a pull-request for this problem. Thanks!